m3/demo/experts/expert_monai_brats.py

```diff
@@ -0,0 +1,125 @@
+
+    def download_file(self, url: str, img_file: str):
+        """
+        Download the file from the URL.
+
+        Args:
+            url (str): The URL.
+            img_file (str): The file path.
+        """
+        parent_dir = os.path.dirname(img_file)
+        os.makedirs(parent_dir, exist_ok=True)
+        with open(img_file, "wb") as f:
+            response = requests.get(url)
+            f.write(response.content)
+
+    def run(
+        self,
+        img_file: list[str] | None = None,
+        image_url: list[str] | None = None,
+        input: str = "",
+        output_dir: str = "",
+        slice_index: int = 0,
+        prompt: str = "",
+        **kwargs,
+    ):
+        """
+        Run the BRATS model.
+
+        Args:
+            image_url (str): The image URL list.
+            input (str): The input text.
+            output_dir (str): The output directory.
+            img_file (str): The image file path list. If not provided, download from the URL.
+            slice_index (int): The slice index.
+            prompt (str): The prompt text from the original request.
+            **kwargs: Additional keyword arguments.
+        """
+        if not img_file:
+            # Download the file from the URL
+            for url in image_url:
+                img_file = os.path.join(output_dir, os.path.basename(url))
+                self.download_file(url, img_file)
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            workflow = create_workflow(
+                workflow_type="infer",
+                bundle_root=self.bundle_root,
+                config_file=os.path.join(self.bundle_root, f"configs/inference.json"),
+                logging_file=os.path.join(self.bundle_root, "configs/logging.conf"),
+                meta_file=os.path.join(self.bundle_root, "configs/metadata.json"),
+                test_datalist=[{"image": img_file}],
+                output_dtype="uint8",
+                separate_folder=False,
+                output_ext=".nii.gz",
+                output_dir=temp_dir,
+            )
+            workflow.evaluator.run()
+            output_file = os.path.join(temp_dir, os.listdir(temp_dir)[0])
+            seg_file = os.path.join(output_dir, "segmentation.nii.gz")
+            move(output_file, seg_file)
+
+        seg_image = f"seg_{uuid4()}.jpg"
+        transforms = get_monai_transforms(
+            ["image", "label"],
+            output_dir,
+            modality="MRI",
+            slice_index=slice_index,
+            axis=2,
+            image_filename=get_slice_filenames(img_file[0], slice_index),
+            label_filename=seg_image,
+        )
+        transforms({"image": img_file[0], "label": seg_file})
+        # TODO: find the right label mapping
+        text_output = f"The results are <segmentation>. The colors in this image describe\nyellow and red: tumor core, only yellow: enhancing tumor, all colors: whole tumor\n"
```

Comment:
Hi @holgerroth , I copied these from 
https://github.com/Project-MONAI/VLM/blob/acdfcf8389205621472c72ebcf37fccd4367172b/m3/data_prepare/experts/expert_utils.py#L98

But I'm unclear what classes does the output label 1, 2, 3?, 4 correspond .

Reply:
The model only cares about the color overlay. That's described in the prompt above.

Reply:
The output should match this description https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation#output

Reply:
Thanks @holgerroth .

Checking the [official data description](https://www.med.upenn.edu/sbia/brats2018/data.html) 

It seems that the following index should be used to produce:
0: background
1: necrotic and no-enhancing tumor core (NCR/NET), which should be counted TC and WT subregion
2: peritumoral edema (ED), which is counted only as WT subresion
4: GD-enhancing tumor (ET), which should be counted under ET, TC, WT subregion

ET: Enhancing Tumor, WT: Whole Tumor, TC: Tumor Core.

I will change this PR to incorporate the right mapping